stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -455,11 +455,12 @@ private void writeRecords() {
     final ActorFuture<Boolean> retryFuture =
         writeRetryStrategy.runWithRetry(
             () -> {
-              final long position = logStreamWriter.tryWrite(pendingWrites, sourceRecordPosition);
-              if (position > 0) {
-                writtenPosition = position;
+              final var writeResult = logStreamWriter.tryWrite(pendingWrites, sourceRecordPosition);
+              if (writeResult.isRight()) {
+                writtenPosition = writeResult.get();
+                return writtenPosition >= 0;
```

Comment:
> This made me think, do we need to accept empty writes? I think I'd slightly prefer if the writer would return an error in that case.

I also thought of returning an error. But I was not sure if it would break anything, because there was a test that verifies the write did not fail.